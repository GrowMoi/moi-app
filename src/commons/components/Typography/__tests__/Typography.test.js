import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components/native';
import { Title, TextBody, Description, Header } from '../index';

describe('Typography', () => {
  it('it should render title expected', () => {
    const wrapper = shallow(<Title>Title</Title>);
    const tree = wrapper.dive();

    expect(tree).toMatchSnapshot();
    expect(tree).toBeTruthy();
  });

  it('it should render header expected', () => {
    const wrapper = shallow(<Header>Header</Header>).dive();

    expect(wrapper.text()).toBe('Header');
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  // TexBody
  it('it should render text body expected', () => {
    const wrapper = renderer.create(<TextBody>Text body</TextBody>);
    wrapper.toJSON();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
  });

  it('it should render text body with props', () => {
    const textInverted = renderer.create(<TextBody inverted>Text body</TextBody>);
    textInverted.toJSON();
    expect(textInverted).toMatchSnapshot();

    const textPurple = renderer.create(<TextBody color='purple'>Text body</TextBody>);
    textPurple.toJSON();
    expect(textPurple).toMatchSnapshot();

    const textSecondary = renderer.create(<TextBody secondary>Text body</TextBody>);
    textSecondary.toJSON();
    expect(textSecondary).toMatchSnapshot();

    const highlightText = renderer.create(<TextBody highlight>Text body</TextBody>);
    highlightText.toJSON();
    expect(highlightText).toMatchSnapshot();
  });

  it('it should render description expected', () => {
    const wrapper = renderer.create(<Description>Description</Description>);
    wrapper.toJSON();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
  });
});
