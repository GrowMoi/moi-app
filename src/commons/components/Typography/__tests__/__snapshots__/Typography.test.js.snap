// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography it should render description expected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "#4a4b21",
        "fontFamily": "Futura",
        "fontSize": 11,
      },
      undefined,
    ]
  }
>
  Description
</Text>
`;

exports[`Typography it should render header expected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "#4a4b21",
        "fontFamily": "Museo300",
        "fontSize": 20,
      },
      undefined,
    ]
  }
>
  Header
</Text>
`;

exports[`Typography it should render text body expected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "#4a4b21",
        "fontFamily": "Futura",
        "fontSize": 15,
      },
      undefined,
    ]
  }
>
  Text body
</Text>
`;

exports[`Typography it should render text body with props 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  inverted={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "rgba(255,255,255,0.9)",
        "fontFamily": "Futura",
        "fontSize": 15,
      },
      undefined,
    ]
  }
>
  Text body
</Text>
`;

exports[`Typography it should render text body with props 2`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  color="purple"
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "purple",
        "fontFamily": "Futura",
        "fontSize": 15,
      },
      undefined,
    ]
  }
>
  Text body
</Text>
`;

exports[`Typography it should render text body with props 3`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  secondary={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "#818180",
        "fontFamily": "Futura",
        "fontSize": 15,
      },
      undefined,
    ]
  }
>
  Text body
</Text>
`;

exports[`Typography it should render text body with props 4`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  highlight={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "#e67543",
        "fontFamily": "Futura",
        "fontSize": 15,
      },
      undefined,
    ]
  }
>
  Text body
</Text>
`;

exports[`Typography it should render title expected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "#4a4b21",
        "fontFamily": "Museo300",
        "fontSize": 25,
        "textAlign": "left",
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;
